tools/iree-benchmark-module-main.cc

```diff
@@ -129,6 +133,54 @@ IREE_FLAG_CALLBACK(
     "Each occurrence of the flag indicates an input in the order they were\n"
     "specified on the command line.");
 
+static iree_status_t parse_time_unit(iree_string_view_t flag_name,
+                                     void* storage, iree_string_view_t value) {
+  auto* unit = (std::pair<bool, benchmark::TimeUnit>*)storage;
+  auto unit_string = std::string(value.data, value.size);
+  if (unit_string == kMillisecondsUnitString) {
+    *unit = {true, benchmark::kMillisecond};
+    return iree_ok_status();
+  } else if (unit_string == kMicrosecondsUnitString) {
+    *unit = {true, benchmark::kMicrosecond};
+    return iree_ok_status();
+  } else if (unit_string == kNanosecondsUnitString) {
+    *unit = {true, benchmark::kNanosecond};
+    return iree_ok_status();
+  }
+  return iree_make_status(IREE_STATUS_INVALID_ARGUMENT,
+                          "unsupported time unit");
+}
+static void print_time_unit(iree_string_view_t flag_name, void* storage,
+                            FILE* file) {
+  auto* unit = (std::pair<bool, benchmark::TimeUnit>*)storage;
+  if (!unit->first) {
+    return;
+  }
+  std::string unit_string;
+  switch (unit->second) {
+    case benchmark::kMillisecond:
+      unit_string = kMillisecondsUnitString;
+      break;
+    case benchmark::kMicrosecond:
+      unit_string = kMicrosecondsUnitString;
+      break;
+    case benchmark::kNanosecond:
+      unit_string = kNanosecondsUnitString;
+      break;
+    default:
+      assert(false && "Unexpected time unit.");
+  }
+  fprintf(file, "--%.*s=\"%s\"\n", (int)flag_name.size, flag_name.data,
+          unit_string.c_str());
+}
+// Time unit to be printed. If the first field is false, each place will use its
+// default time unit.
+static std::pair<bool, benchmark::TimeUnit> FLAG_time_unit = {
+    false, benchmark::kNanosecond};
```

Comment:
Seems like that could be recorded via the empty string? Then we don't need a pair or anything

Reply:
I prefer to convert the string value into enum when parsing the flag. So we don't need to parse twice later. Really hope I have std::optional to use...